### DISCUSSION TITLE
GraphQL fragments ?

### DISCUSSION THREAD

**Discussion by MSP-Greg** [community/community#24431] [Apps, API and Webhooks]
<p>Just wondering there is an expectation that fragments will be implemented?</p>
<p>I couldn’t find mention of them anywhere, or in the graphql-client gem…</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>I ran this query just now:</p>
<pre><code>query {
  repository(owner: "wojtekmaj" name:"react-date-picker"){
    object(oid:"50e933af2f28ca8da451c63afcced8dca467f2c7"){
      ...fields
    }
  }
}

fragment fields on Commit {
  oid
  changedFiles
  commitResourcePath
  commitUrl
  message
  resourcePath
  url
  tree{
    entries{
      name
    }
  }
}
</code></pre>
<p>And the fragment seemed to work. What is it about fragments that isn’t working for you?</p>

**Comment by MSP-Greg** [comment]
<p>Apologies, my bad.</p>
<p>I was looking here and there (ref material and code) and thought fragments were similar to variables, in that they were added to the json hash with a key of fragments.  As you’ve shown, they are a part of the query value…</p>

**Comment by lee-dohm** [comment]
<p>No apologies necessary. I’m learning this as I go too 😀</p>
